### PR TITLE
Change the custom error to Ballerina error

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
Change the custom error to Ballerina error

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: consul/consul_connector.bal [hunk 1]
```diff
@@ -446,16 +396,25 @@ function ConsulConnector::deleteKey(string keyName) returns (boolean|ConsulError
...
-                        return consulError;
+                        return setStringResponseError(stringResponse);
                     }
                 }
             }
         }
     }
 }
+
+function setJsonResponseError(json response) returns error {
```
Comment (Biruntha) [inline consul/consul_connector.bal]: Move these methods to the file where we write other utils methods

File: README.md [hunk 2]
```diff
@@ -27,8 +27,8 @@ git clone https://github.com/wso2-ballerina/module-consul.git
 
 ### Working with Consul Connector actions
 
-All the actions return valid response or ConsulError. If the action is a success, then the requested resource will 
-be returned. Else ConsulError object will be returned.
+All the actions return valid response or error. If the action is a success, then the requested resource will 
+be returned. Else error object will be returned.
```
Comment (VijithaEkanayake) [inline README.md]: Just say error will be returned
```suggestion
be returned. Else error will be returned.
```

File: consul/Module.md [hunk 3]
```diff
@@ -56,7 +56,7 @@ Register services in Consul with the given `jsonPayload`.
 var serviceRegister = consulClient->registerService(jsonPayload);
 match serviceRegister {
     boolean response => io:println(response);
-    consul:ConsulError err => io:println(err);
+    consul:error err => io:println(err);
```
Comment (VijithaEkanayake) [inline consul/Module.md]:
```suggestion
    error err => io:println(err);
```
Check all the relevant places

Comment (kalaiyarasiganeshalingam) [reply consul/Module.md]: Fixed.

File: README.md (hunk 2 above)
Comment (kalaiyarasiganeshalingam) [reply README.md]: Fixed

**Review by NipunaRanasinghe — APPROVED** [review]

**Review by VijithaEkanayake-zz — APPROVED** [review]

**Pull request merged**